tests/api_fastapi/core_api/routes/public/test_xcom.py

```diff
@@ -0,0 +1,226 @@
+            "key": TEST_XCOM_KEY,
+            "task_id": TEST_TASK_ID,
+            "map_index": -1,
+            "timestamp": current_data["timestamp"],
+            "value": str(TEST_XCOM_VALUE),
+        }
+
+    def test_should_respond_200_native(self, test_client):
+        self.create_xcom(TEST_XCOM_KEY, TEST_XCOM_VALUE)
+        response = test_client.get(
+            f"/public/dags/{TEST_DAG_ID}/dagRuns/{run_id}/taskInstances/{TEST_TASK_ID}/xcomEntries/{TEST_XCOM_KEY}?stringify=false"
+        )
+        assert response.status_code == 200
+
+        current_data = response.json()
+        assert current_data == {
+            "dag_id": TEST_DAG_ID,
+            "execution_date": execution_date_parsed.strftime("%Y-%m-%dT%H:%M:%SZ"),
+            "key": TEST_XCOM_KEY,
+            "task_id": TEST_TASK_ID,
+            "map_index": -1,
+            "timestamp": current_data["timestamp"],
+            "value": TEST_XCOM_VALUE,
+        }
+
+    @conf_vars({("core", "enable_xcom_pickling"): "True"})
+    def test_should_respond_200_pickled(self, test_client):
+        self.create_xcom(TEST_XCOM_KEY2, TEST_XCOM_VALUE2)
+        response = test_client.get(
+            f"/public/dags/{TEST_DAG_ID}/dagRuns/{run_id}/taskInstances/{TEST_TASK_ID}/xcomEntries/{TEST_XCOM_KEY2}"
+        )
+        assert response.status_code == 200
+
+        current_data = response.json()
+        assert current_data == {
+            "dag_id": TEST_DAG_ID,
+            "execution_date": execution_date_parsed.strftime("%Y-%m-%dT%H:%M:%SZ"),
+            "key": TEST_XCOM_KEY2,
+            "task_id": TEST_TASK_ID,
+            "map_index": -1,
+            "timestamp": current_data["timestamp"],
+            "value": str(TEST_XCOM_VALUE2),
+        }
+
+    def test_should_raise_404_for_non_existent_xcom(self, test_client):
+        response = test_client.get(
+            f"/public/dags/{TEST_DAG_ID}/dagRuns/{run_id}/taskInstances/{TEST_TASK_ID}/xcomEntries/{TEST_XCOM_KEY3}"
+        )
+        assert response.status_code == 404
+        assert response.json()["detail"] == f"XCom entry with key: `{TEST_XCOM_KEY3}` not found"
+
+    @pytest.mark.parametrize(
+        "support_deserialize, params, expected_status_or_value",
+        [
+            pytest.param(
+                True,
+                "?deserialize=true",
+                f"real deserialized {TEST_XCOM_VALUE}",
+                id="enabled deserialize-true",
+            ),
+            pytest.param(
+                False,
+                "?deserialize=true",
+                400,
+                id="disabled deserialize-true",
+            ),
+            pytest.param(
+                True,
+                "?deserialize=false",
+                f"orm deserialized {TEST_XCOM_VALUE}",
+                id="enabled deserialize-false",
+            ),
+            pytest.param(
+                False,
+                "?deserialize=false",
```

Comment:
We can use a dict instead, and pass that to `test_client.get(..., params=params)`, instead of doing the url encoding ourselves